### ISSUE TITLE
caddy hot reload config

### ISSUE THREAD

**Issue by ofsahin** [lucaslorentz/caddy-docker-proxy#243]
hello,

i just started using this plugin and it is awesome 🤞😁
thanks for the help @francislavoie 

now the question
normally when i change caddyfile, i hot reload the config 
docker exec -it caddy /usr/bin/caddy reload --config /etc/caddy/Caddyfile --adapter caddyfile
but since i started using this plugin, i could not figure out how to reload the base caddyfile set by -caddyfile-path

i tried
docker exec -it caddy /usr/bin/caddy docker-proxy reload -caddyfile-path /etc/caddy/Caddyfile

but when i use it like this, i end up in the container and cursor stays there, if i ctrl+c it says
INFO    shutting down   {"signal": "SIGINT"}
WARN    exiting; byeee!! 👋     {"signal": "SIGINT"}
INFO    shutdown complete       {"signal": "SIGINT", "exit_code": 0}

is there a way to hot reload base caddyfile?

**Comment by francislavoie** [comment]
I don't think this is possible right now, it would probably require a new command to work. I think the Caddyfile is just read once when starting `docker-proxy` and never again.

You should probably move the things you have in that Caddyfile into labels on your Caddy service though.

**Comment by lucaslorentz** [comment]
Or you could move it to docker swarm config to keep using the Caddyfile syntax

**Comment by MagicJF** [comment]
Is there a way to view the operating caddyfile?

**Comment by francislavoie** [comment]
@MagicJF no need to make a comment in another issue after having opened one. https://github.com/lucaslorentz/caddy-docker-proxy/issues/502 Your question is not relevant to this feature request.

**Comment by pedroapero** [comment]
> The Caddyfile is automatically reloaded by polling (default every 30 seconds, see CADDY_DOCKER_POLLING_INTERVAL), so there shouldn't be any need to manually reload or even restart. Works for me.
> 
> One issue I faced in the beginning: If you mounted just the Caddyfile into the container and not the config directory, you must edit the file in-place. Editors like e.g. VIM write a new file and rename it, so it gets a new inode and won't be picked up by the container. Two proposals to work around that:
> 
>     1. mount the directory
> 
>     2. create a symlink to the Caddyfile and point your editor to edit the symlink. This will trigger in-place update of the file which can be seen by the bind-mount.

Hello, I can see that Caddy is reloading itself but I fail to understand how it could find the docker-compose configuration by itself. I am running the docker-compose binary; if I edit the docker-compose.yaml nothing gets reloaded automatically.

Running `docker-compose up --no-recreate` does not reload the Caddy configuration.
Restarting containers (`--force-recreate`) does works but I obviously don't want to keep restarting my containers when I need to test proxy options.

**Comment by pedroapero** [comment]
> The Caddyfile is automatically reloaded by polling (default every 30 seconds, see CADDY_DOCKER_POLLING_INTERVAL), so there shouldn't be any need to manually reload or even restart. Works for me.
> 
> One issue I faced in the beginning: If you mounted just the Caddyfile into the container and not the config directory, you must edit the file in-place. Editors like e.g. VIM write a new file and rename it, so it gets a new inode and won't be picked up by the container. Two proposals to work around that:
> 
>     1. mount the directory
> 
>     2. create a symlink to the Caddyfile and point your editor to edit the symlink. This will trigger in-place update of the file which can be seen by the bind-mount.

Hello, I can see that Caddy is reloading itself but I fail to understand how it could find the docker-compose configuration by itself. I am running the docker-compose binary; if I edit the docker-compose.yaml nothing gets reloaded automatically.

Running `docker-compose up --no-recreate` does not reload the Caddy configuration.
Restarting containers (`--force-recreate`) does works but I obviously don't want to keep restarting my containers when I need to test proxy options.

The documentation says
> Every time a Docker object changes, the plugin updates the Caddyfile

How can I trigger such a change?

**Comment by francislavoie** [comment]
It's not the docker-compose files that it watches, it's the docker labels, which are attached to the actual running containers in memory (i.e. `docker inspect <container-id>`, you'll see the labels).

You must re-create containers for label changes to take effect.

**Comment by pedroapero** [comment]
I can indeed see the labels not getting updated without restarting the container.
Thanks for your reply!

**Comment by polarathene** [comment]
> I obviously don't want to keep restarting my containers when I need to test proxy options.

Probably not what you want either but if restarting implicitly due to local file changes (_might apply to compose file, I haven't tested_), you could try compose watch?: https://docs.docker.com/compose/file-watch/

**Comment by lucaslorentz** [comment]
This already works — the base Caddyfile passed via `--caddyfile-path` / `CADDY_DOCKER_CADDYFILE_PATH` is re-read from disk on every generator iteration (see [`generator/generator.go:75`](https://github.com/lucaslorentz/caddy-docker-proxy/blob/master/generator/generator.go#L75)), so edits are picked up automatically:

- On each Docker event (container/service/config/network changes), or
- On each polling tick (default `30s`, configurable via `CADDY_DOCKER_POLLING_INTERVAL`)

No manual reload command is needed. @bb described this correctly in [this comment](https://github.com/lucaslorentz/caddy-docker-proxy/issues/243#issuecomment-939765446).

**Gotcha to be aware of** (from the same comment): if you bind-mount just the Caddyfile (not its parent directory), editors like `vim` write to a new inode and rename, which breaks the mount. Workarounds:
1. Mount the parent directory instead of the file, or
2. Edit via a symlink so the underlying inode stays the same.

Closing as working as designed. If you need a force-reload command for a specific scenario (e.g. to avoid waiting for the polling interval), please open a new issue describing the use case.

_— Closed by Claude (AI agent) on behalf of @lucaslorentz._
